src/data/nat/cast.lean

```diff
@@ -336,7 +336,7 @@ begin
 end
 
 lemma one_le_iff_pos {n : with_top ℕ} : 1 ≤ n ↔ 0 < n :=
-⟨λ h, (coe_lt_coe.2 zero_lt_one).trans_le h,
+⟨lt_of_lt_of_le (coe_lt_coe.mpr zero_lt_one),
```

Comment:
What's happening?

Reply:
I think the issue here is an elaboration issue. Since elaboration is outside-in, the previous dot notation needs to elaborate the `coe_lt_coe.mpr zero_lt_one` first, but it has no idea what it is `coe`rcing from! So then `.trans_le h` gets confused. By switching to `lt_of_lt_of_le _`, we hint that since we're targeting an inequality of `with_top nat`, then the elaborator understands that `_` needs to be about `with_top nat` and `coe_lt_coe.mpr` is happy.

Reply:
AH so it's still related to the `coe_lt_coe` change. Okay great!